### PR TITLE
Lint magic numbers / error in lint message

### LINKED ISSUE #2917: Error in lint message
When giving meta data `// @` without a name, `"version"` is falsely replaced by `""` in the warning message, because it continues with `but end of input found.` (not `but "<something>" found.`).

Wrong message:
```
/!\ Expected "author", "description", "downloadURL", "exclude", "grant",
"homepageURL", "icon", "include", "installURL", "match", "name",
"namespace", "noframes", "require", "resource", "run-at", "updateURL" or
"" but end of input found.
```

Correct message:
```
/!\ Expected "author", "description", "downloadURL", "exclude", "grant",
"homepageURL", "icon", "include", "installURL", "match", "name",
"namespace", "noframes", "require", "resource", "run-at", "updateURL" or
"version" but end of input found.
```

### LINKED ISSUE #2918: Magic number 4 in lint: META_POSN
Minor issue in `lint-metadata.js`:
`META_POSN = 4` is defined in lint handler, but not entirely used there.
Also `// @` (the prefix, that is responsible for that number 4) is given more than once.

### PR DESCRIPTION
**Pull request by Eselce**
Lint handling issues in meta data block...

Fixes #2918
Fixes #2917

### CODE REVIEW

File: src/content/cm-addons/lint-metadata.js [hunk 1]
```diff
@@ -15,21 +16,21 @@ function lintMetadata(text, options) {
   for (let [i, metaLine] of meta.entries()) {
     try {
       // Only consider meta lines.
-      if (metaLine.substr(0, 4) == '// @') {
+      if (metaLine.substr(0, META_POSN) === META_PREFIX) {
         parseMetaLine(metaLine.trim());
```
Comment (arantius) [inline src/content/cm-addons/lint-metadata.js]: Just `metaLine.startsWith(META_PREFIX)` ?

File: src/content/cm-addons/lint-metadata.js [hunk 2]
```diff
@@ -15,21 +16,21 @@ function lintMetadata(text, options) {
...
       }
     } catch (e) {
       let ann = {
-        severity: 'warning',
-        from: { line: metaStartLine + i, ch: e.location.start.column - 1 },
-        to: { line: metaStartLine + i, ch: e.location.end.column - 1 },
-        message: e.message,
```
Comment (arantius) [inline src/content/cm-addons/lint-metadata.js]: Always include optional trailing comma.

https://greasemonkey.github.io/style.html

File: src/content/cm-addons/lint-metadata.js [hunk 3]
```diff
@@ -15,21 +16,21 @@ function lintMetadata(text, options) {
...
+        'from': { line: metaStartLine + i, ch: e.location.start.column - 1 },
+        'to': { line: metaStartLine + i, ch: e.location.end.column - 1 },
+        'message': e.message
       };
       // Turn "Expected...but 'w' found" into "Expected...but 'word' found".
-      if (e.message.match(/Expected.*but.*found/) && ann.from.ch == META_POSN) {
-        let m = metaLine.match(/^\/\/ @([^\s]*)/);
+      if (e.message.match(/Expected .* but .* found\./) && ann.from.ch === META_POSN) {
+        let m = metaLine.substr(META_POSN).match(/^(\s+|\S*)/);
+        ann.message = ann.message.replace(/( but ")([^"]*)(" found\.)$/, `$1${m[1]}$3`);
```
Comment (arantius) [inline src/content/cm-addons/lint-metadata.js]: Line break here, 80 character max.

File: src/content/cm-addons/lint-metadata.js (hunk 1 above)
Comment (Eselce) [reply src/content/cm-addons/lint-metadata.js]: Definitely better!

File: src/content/cm-addons/lint-metadata.js (hunk 2 above)
Comment (Eselce) [reply src/content/cm-addons/lint-metadata.js]: Done.

File: src/content/cm-addons/lint-metadata.js (hunk 3 above)
Comment (Eselce) [reply src/content/cm-addons/lint-metadata.js]: Oops, trying to respect 80 chars, sometimes it fails...

**Pull request merged**